plugins/org.eclipse.n4js/src/org/eclipse/n4js/parser/RegExLiteralAwareLexer.java

```diff
@@ -128,6 +127,23 @@ public Token nextToken() {
 				throw new RuntimeException("Unexpected recognition problem for\n" + input, re);
 			}
 		} else {
+			// check for a sequence `?.[digit]` which must not yield `?.`. `[digit]` but `?` `.[digit]`
+			if (input.LA(1) == '?' && input.LA(2) == '.') {
+				int propablyDigit = input.LA(3);
+				if (propablyDigit >= '0' && propablyDigit <= '9') {
+					try {
+						clearAndResetTokenState();
+						mQuestionMark();
+						if (this.state.token == null) {
+							emit();
+						}
+						return this.state.token;
+					} catch (RecognitionException re) {
+						// this is basically impossible since the regex tail consumes any char except for line breaks
+						throw new RuntimeException("Unexpected recognition problem for\n" + input, re);
+					}
+				}
+			}
 			return super.nextToken();
```

Comment:
Great the lexer can be customized in this way...
Infinite possibilities! :stuck_out_tongue_winking_eye: